identify/identify.py

```diff
@@ -159,6 +159,15 @@ def parse_shebang(bytesio):
     cmd = tuple(_shebang_split(first_line.strip()))
     if cmd and cmd[0] == '/usr/bin/env':
         cmd = cmd[1:]
+    if cmd and cmd[0].endswith('nix-shell'):
+        if bytesio.read(2) != b'#!':
+            return cmd
+        second_line = bytesio.readline().decode('UTF-8')
```

Comment:
will need an additional check for non-UTF-8 and non-readable as above

Reply:
fixed in https://github.com/blaggacao/identify/commit/1595423afadb8c3f137b2371c31a9abcf7f1d6c7